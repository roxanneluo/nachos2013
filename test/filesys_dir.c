#include "filesysgrader.h"
#include "stdio.h"
#include "stdlib.h"

char buf[100];

int main(int argc, char** argv)
{
    int size = getFreeDiskSize();
    assertTrue(!mkdir("dir1"));
    assertTrue(!mkdir("dir1/dir1_1"));
    printCwd();
    chdir("dir1/dir1_1");
    printCwd();
    chdir("../../dir1/dir1_1/.././..");// /dir1/dir1_1/../../dir1/dir1_1/.././..==/
    printCwd();
    mkdir("dir2");// mkdir /dir2
    chdir("dir2");// cwd /dir2
	//printCwd();
    //assertTrue(rmdir("/dir1")); // for dir1_1 in /dir1 corrected editor:MLSheng
    assertTrue(rmdir("dir1"));  // test path added by MLSheng
	//printCwd();
    assertTrue(rmdir("/dir1/dir1"));//of couse only /dir1/dir1_1 exists
	//printCwd();
	assertTrue(!rmdir("/dir1/dir1_1"));// or size is not correct corrected editor:MLSheng
	//printCwd();
	assertTrue(!rmdir("/dir1"));// or size is not correct corrected editor:MLSheng
    printCwd();// /dir2
    chdir("/");// cwd /
    assertTrue(!rmdir("/dir2"));
    assertTrue(getFreeDiskSize() == size);
    done();
    return 0;
}
