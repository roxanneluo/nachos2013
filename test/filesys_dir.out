/
/dir1/dir1_1
/
/dir2
